docs/source/configuration/overview.mdx

```diff
@@ -1175,18 +1175,27 @@ example:
 
 ### Fragment reuse and generation
 
-By default, the router will attempt to reuse fragments from the original query while forming subgraph requests. This behavior can be disabled by setting the option to `false`:
+By default, the router will _generate_ fragments for subgraph requests,
+extracting _inline fragments only_ into fragment definitions before sending
+queries to subgraphs. In very many cases this significantly reduces the size of
+the query sent to subgraphs, but may increase the time it takes for planning.
+Note that this option and `experimental_reuse_query_fragments` are mutually
+exclusive; if both are explicitly set to `true`, `generate_query_fragments` will
+take precedence. This behaviour can be disabled by setting the option to `false`:
```

Comment:
I wouldn't emphasise too much what exactly it does (wrt inline fragment extraction), as we may want to improve the algorithm in the future, for example also generating fragments for repeating patterns in field selections.

We should not say it increases the time it takes for planning. Generating query fragments takes a trivial amount of time compared to fragment reuse.

Perhaps something along these lines:

```suggestion
By default, the router will compress subgraph requests by generating fragment
definitions based on the shape of the subgraph operation. In many cases this
significantly reduces the size of the query sent to subgraphs.
Note that this option and `experimental_reuse_query_fragments` are mutually
exclusive; if both are explicitly set to `true`, `generate_query_fragments` will
take precedence. This behaviour can be disabled by setting the option to `false`:
```